### PR TITLE
Operation: Extract Hash Values

### PR DESCRIPTION
**Pull request by MShwed**
References https://github.com/gchq/CyberChef/issues/466

Adds operation for extracting hash values from text blocks. Searches text for hashes based on length.  
Options for:
- Searching for hashes of all lengths. Potential hash lengths taken from "Analyse Hashes" operation. 
- Displaying total hashes found.

Screen Shot 2019-03-11 at 20 01 24

If it makes more sense to select hash types by name from a dropdown rather than by character length, I can change that.

### CODE REVIEW

**Comment by a3957273** [comment]
Hey, would you be able to resolve the conflicts and update this to the latest version of master?  Would be good to get this merged in :)

**Comment by MShwed** [comment]
No problem! I'll get that resolved

File: src/core/operations/ExtractHashes.mjs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        if (searchAllHashes) hashBitLengths = [4, 8, 16, 32, 64, 128, 160, 192, 224, 256, 320, 384, 512, 1024];
+
+        for (const hashBitLength of hashBitLengths) {
+            // Convert bit length to character length
+            const hashCharacterLength = (hashBitLength / 8) * 2;
+
+            const regex = new RegExp(`(\\b|^)[a-f0-9]{${hashCharacterLength}}(\\b|$)`, "g");
+            const searchResults = search(input, regex, null, false);
+
+            hashCount += searchResults.split("\n").length - 1;
```
[the remark this answers is not in the document]
Comment (MShwed) [reply src/core/operations/ExtractHashes.mjs]: Thank you! That works for me as well.

File: src/core/operations/ExtractHashes.mjs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+ * @license Apache-2.0
+ */
+
+import Operation from "../Operation.mjs";
+import { search } from "../lib/Extract.mjs";
+
+/**
+ * Extract Hash Values operation
+ */
+class ExtractHashes extends Operation {
```
[the remark this answers is not in the document]
Comment (MShwed) [reply src/core/operations/ExtractHashes.mjs:13]: No problem! I've added some unit tests.

**Review by a3957273 — APPROVED** [review]

**Comment by a3957273** [comment]
Looks great to me, thanks for continuing to contribute when you originally opened your PR over 5 years ago!

**Pull request merged**